geomesa-accumulo/geomesa-accumulo-tools/src/test/scala/org/locationtech/geomesa/accumulo/tools/ingest/IngestCommandTest.scala

```diff
@@ -9,210 +9,223 @@
       command.execute()
 
-      val features = command.withDataStore { ds =>
-        SelfClosingIterator(ds.getFeatureSource("renegades").getFeatures.features).toList
+      command.withDataStore { ds =>
+        try {
+          val features = SelfClosingIterator(ds.getFeatureSource("renegades").getFeatures.features).toList
+          features must haveSize(5)
+          features.map(_.getAttribute("name")) must
+              containTheSameElementsAs(Seq("Hermione", "Harry", "Severus", "Ron", "Ginny"))
+        } finally {
+          ds.delete()
+        }
       }
-      features.size mustEqual 5
-      features.map(_.get[String]("name")) must containTheSameElementsAs(Seq("Hermione", "Harry", "Severus", "Ron", "Ginny"))
     }
 
-    "work with mock with provided ZK and instance and default accumulo passwords!" >> {
-      val id = nextId
-
-      val conf = ConfigFactory.load("examples/example1-csv.conf")
-      val sft = conf.root().render(ConfigRenderOptions.concise())
-      val converter = conf.root().render(ConfigRenderOptions.concise())
-      val dataFile = new File(this.getClass.getClassLoader.getResource("examples/example1.csv").getFile)
-
-      val combinations =
-        Seq (
-          (Array("-z", "foobar", "-i", "hello"), 17)
-        )
-
-      val userCombinations = Seq (
-        ("myuser", "mypassword"),
-        ("root", "secret"), // default
-        ("root", "other")  // default user
-      )
-
-      combinations.zip(userCombinations).forall { case ((additionalArgs, count), (user, pass)) =>
-        val testArgs = Array("ingest", "--mock", "--user", user, "--password", pass)
-
-        val args = testArgs ++ Array("--catalog", id, "--converter", converter, "-s", sft, dataFile.getPath) ++ additionalArgs
-
-        args.length mustEqual count
-
-        val command = AccumuloRunner.parseCommand(args).asInstanceOf[AccumuloDataStoreCommand]
-        command.execute()
-
-        val features = command.withDataStore(ds => SelfClosingIterator(ds.getFeatureSource("renegades").getFeatures.features).toList)
-        features.size mustEqual 3
-        features.map(_.get[String]("name")) must containTheSameElementsAs(Seq("Hermione", "Harry", "Severus"))
-      }
+    "fail without user" >> {
+      val dataFile = new File(getClass.getClassLoader.getResource("examples/example1.csv").getFile)
+      val base = baseArgs
+      val i = base.indexOf("--user")
+      val args = base.slice(0, i) ++ base.slice(i + 2, base.length) ++ Array(dataFile.getPath)
+      AccumuloRunner.parseCommand(args) must throwA[com.beust.jcommander.ParameterException]
+    }
 
+    "fail with user and password and keytab" >> {
+      val dataFile = new File(getClass.getClassLoader.getResource("examples/example1.csv").getFile)
+      val args = baseArgs ++ Array("--keytab", "/path/to/some/file", dataFile.getPath)
+      AccumuloRunner.parseCommand(args) must throwA[com.beust.jcommander.ParameterException]
     }
 
+    // TODO get kerberos tests working
```

Comment:
Should we update this to reference GEOMESA-2797 as well?  (or vice versa)

Reply:
added